src/dlgPackageExporter.cpp

```diff
@@ -268,64 +317,230 @@ void dlgPackageExporter::slot_export_package()
+                               << "it is a Symlink - avoided to prevent file-system loops!";
+                    continue;
+                }
+
+                QString nameInArchive = itDir.filePath();
+                nameInArchive.remove(QStringLiteral("%1/").arg(mStagingDirName));
+
+                if       (entryInfo.isDir()) {
+                    directoryEntries.append(nameInArchive);
+                } else if(entryInfo.isFile()) {
+                    fileEntries.insert(nameInArchive, itDir.filePath());
+                }
+            }
+
+#if defined(Q_OS_WIN32)
+            qt_ntfs_permission_lookup--;
+#endif
+
+            if (directoryEntries.count() > 1) {
+                std::sort(directoryEntries.begin(), directoryEntries.end());
+            }
+
+            QStringListIterator itDirectoryName(directoryEntries);
+            while (itDirectoryName.hasNext() && isOk) {
+                QString directoryName = itDirectoryName.next();
+                // zip_dir_add(...) returns the index of the
+                // added item in the archive or -1 on error:
+                if (zip_dir_add(archive, directoryName.toStdString().c_str(), ZIP_FL_ENC_UTF_8) == -1) {
+                    displayResultMessage(tr("Failed to add directory \"%1\" to module (archive) file, error is: \"%2\".")
+                                         .arg(directoryName, QString::fromUtf8(zip_strerror(archive))), false);
+                    zip_close(archive);
+                    isOk = false;
+                }
+            }
+
+            // Process the config and the file containing the Mudlet triggers,
+            // etc. specially so they are inserted first and last respectively:
+            if (isOk) {
+                // Apparently it is permissable for there NOT to be a config.lua
+                // file in the manufactured module (i.e. the user is allowed to
+                // manually remove it even though we now request they do not).
+                // If we enhance modules in the future to store more data in
+                // that file I think this will no longer be permissable and the
+                // two commented out bits of code can be restored
+                if (fileEntries.contains(QStringLiteral("config.lua"))) {
+                    if (!writeFileToZip(QStringLiteral("config.lua"), fileEntries.value(QStringLiteral("config.lua")), archive)) {
+                        /* isOk = false; */
+                    } else {
+                        fileEntries.remove(QStringLiteral("config.lua"));
+                    }
+                }/* else {
+                    displayResultMessage(tr("Required \"config.lua\" file not found to include in module (archive) file, did you remove or rename it?"), false);
+                    zip_close(archive);
+                    isOk = false;
+                }*/
+            }
+
+            QString xmlFileName = mPackageName;
+            xmlFileName.append(QLatin1String(".xml"));
```

Comment:
Not sure why these lines are sprinkled in between all those `if (isOk)` constructs. Easy to overlook.

Reply:
This declaration has to be outside the following `if (`...`)` because it may also used in a second `if (`...`)` *immediately* afterwards - however it is not needed *elsewhere* - so it is being kept as local to its usage as it can...